proto/redpanda/core/admin/cluster_link.proto

```diff
@@ -0,0 +1,409 @@
+}
+// TLS file settings
+message TLSFileSettings {
+    // Path to the CA
+    string ca_path = 1;
+    // Key and Cert are optional but if one is provided, then both must be
+    // Path to the key
+    string key_path = 2;
+    // Path to the cert
+    string cert_path = 3;
+}
+// Used when providing the TLS information in PEM format
+message TLSPEMSettings {
+    // The CA
+    string ca = 1;
+    // Key and Cert are optional but if one is provided, then both must be
+    // The key
+    string key = 2 [(google.api.field_behavior) = INPUT_ONLY];
+    // Indicates that the key has been set
+    bool key_set = 3 [(google.api.field_behavior) = OUTPUT_ONLY];
+    // The cert
+    string cert = 4;
+}
+// SCRAM settings
+message ScramConfig {
+    // SCRAM username
+    string username = 1;
+    // Password
+    string password = 2 [(google.api.field_behavior) = INPUT_ONLY];
+    // Indicates that the password has been set
+    bool password_set = 3 [(google.api.field_behavior) = OUTPUT_ONLY];
+    // The SCRAM mechanism to use
+    ScramMechanism scram_mechanism = 4;
+}
+// A filter based on the name of a resource
+message NameFilter {
+    // Literal or prefix
+    PatternType pattern_type = 1;
+    // Include or exclude
+    FilterType filter_type = 2;
+    // The resource name, or "*"
+    // Note if "*", must be the _only_ character
+    // and `pattern_type` must be `PATTERN_TYPE_LITERAL`
+    string name = 3;
+}
+// A filter for ACLs
+message ACLFilter {
+    // The resource filter
+    ACLResourceFilter resource_filter = 1;
+    // The access filter
+    ACLAccessFilter access_filter = 2;
+}
+// A filter to match ACLs for resources
+message ACLResourceFilter {
+    // The ACL resource type to match
+    common.ACLResource resource_type = 1;
+    // The pattern to apply to name
+    common.ACLPattern pattern_type = 2;
+    // Name, if not given will default to match all items in `resource_type`.
+    // Note that asterisk `*` is literal and matches resource ACLs
+    // that are named `*`
+    string name = 3;
+}
+// Filter an ACL based on its access
+message ACLAccessFilter {
+    // The name of the principal, if not set will default to match
+    // all principals with the specified `operation` and `permission_type`
+    string principal = 1;
+    // The ACL operation to match
+    common.ACLOperation operation = 2;
+    // The permission type
+    common.ACLPermissionType permission_type = 3;
+    // The host to match.  If not set, will default to match all hosts
+    // with the specified `operation` and `permission_type`. Note that
+    // the asterisk `*` is literal and matches hosts that are set to `*`
+    string host = 4;
+}
+// Status of the cluster link
+message ClusterLinkStatus {
+    ClusterLinkState state = 1;
+    // Statuses of the running tasks
+    repeated TaskStatus task_status = 2;
+    // Status of mirror topics
+    repeated MirrorTopicStatus mirror_topic_status = 3;
+}
+// Status of a task
+message TaskStatus {
```

Comment:
Is this TaskStatus specific for cluster links?
If yes, it should have a `ClusterLink` prefix like `ClusterLinkTaskStatus`, or it might be confusing because it's going to be a top level member of the admin namespace.